packages/application-shell/src/components/navbar/navbar.js

```diff
@@ -242,14 +244,21 @@ const hasMenuEveryMenuVisibilitySetToBeHidden = (
   namesOfMenuVisibilities.every(
     nameOfMenuVisibility => menuVisibilities[nameOfMenuVisibility] === true
   );
+const isDisabledForEnvironment = (uriOfMenu, environment) => {
+  const nameOfEnvironmentKey = camelCase(`disable-${slugify(uriOfMenu)}`);
+  const valueOfEnvironmentKey = environment[nameOfEnvironmentKey];
+
+  return valueOfEnvironmentKey === true || valueOfEnvironmentKey === 'true';
+};
 export const RestrictedMenuItem = props => {
   // NOTE: Custom application are activated/deactivated while their
   // visibility is not controlled via a visibiility overwrite.
   if (
     hasMenuEveryMenuVisibilitySetToBeHidden(
       props.namesOfMenuVisibilities,
       props.menuVisibilities
-    )
+    ) ||
+    isDisabledForEnvironment(props.uriPath, props.environment)
```

Comment:
This simplicity of this part is okay.

Reply:
I would suggest to use the `key` and not the `uriPath`. The `key` is usually already the slugified `uriPath`.

Also, how about passing a list of menu item keys to be disabled?

```json
{
  "disabledMenuItems": ["settings-project", "settings-product-types"]
}
```